### PR TITLE
warpx: only allow master branch, and fix build on darwin

### PR DESCRIPTION
**Pull request by dpgrote**
These changes are needed so that WarpX will build without needing any extra specifications.

### CODE REVIEW

**Comment by ax3l** [comment]
Why is there no OpenMP on darwin? Might that be compiler specific?

**Comment by dpgrote** [comment]
On the mac, the default clang (and gcc which is linked to clang) do not support openmp. I don't know enough spack to know how to be more specific. Is there a way to check if a compiler supports openmp? Is there a way to check if the compiler being used is the built in one?

The problem with the dev branch is that it needs the development branch of amrex. Is there a way to modify the resource so that it uses the master branch with the master version and development branch with the dev version?

**Comment by ax3l** [comment]
> On the mac, the default clang (and gcc which is linked to clang) do not support openmp. [...]
> Is there a way to check if a compiler supports openmp?

Good to know!
You can probably probe `self.compiler.openmp_flag` for that :)

Also, you might want to add a variant for OpenMP, defaulting to `variant("openmp", default=(sys.platform != 'darwin')` for convenience and higher control.

> The problem with the dev branch is that it needs the development branch of amrex.

I see! So the thing in spack is the following: as long as you do not add a numbered version, the default order is `develop`, `master`, `head`, `trunk` and then all other "named" version.

So I would suggest the following: keep `develop` but set `master` as `preferred=True` in the version.

Then, add `depends_on('amrex', when='@develop')` without any constrain on amrex. That way, a developer that tries to `spack install warpx@develop` will first get the amrex defaults and in case that version does not work (aka: last stable) the power-user/developer can just add further constrains, such as `spack install warpx@develop ^amrex@develop`. That gives most flexibility, I would say (because assuming that warpx develop always needs the latest amrex develop would also be a constrain we can leave open, I guess).

File: var/spack/repos/builtin/packages/warpx/package.py [hunk 1]
```diff
@@ -38,14 +37,21 @@ class Warpx(MakefilePackage):
...
 
     resource(name='picsar',
              git='https://bitbucket.org/berkeleylab/picsar.git',
              tag='master',
              destination='.')
 
+    @property
+    def build_targets(self):
+        if self.spec.satisfies('%clang'):
+            return ['CXXFLAGS=-std=c++11']
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/warpx/package.py]: better use 
```suggestion
            return ['CXXFLAGS={0}'.format(self.compiler.cxx11_flag)]
```

for broader compiler support, e.g. with older clang versions

**Comment by dpgrote** [comment]
I've addressed various issues. Two comments:

I could only get the UnsupportedCompilerFlag exception by explicitly importing it. I see other packages use it (clingo and p4est) without the explicit import. Are they broken or is there something else that needs to happen in this package.py?

This depends on issue #11666 being addressed.

**Comment by ax3l** [comment]
I haven't used `UnsupportedCompilerFlag` yet, but looks ok to me. Potentially that's really missing in those other packages (or in the general import) as well.

**Comment by dpgrote** [comment]
This latest change gets around the problem with the resources. This version works with both the develop and master branches and is ready to be merged in.

**Pull request merged**